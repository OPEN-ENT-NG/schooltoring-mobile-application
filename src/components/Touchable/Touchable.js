import React from "react";
import {
  TouchableNativeFeedback,
  TouchableOpacity,
  Platform
} from "react-native";
import PropTypes from "prop-types";

const Touchable = props => {
  return Platform.OS === "android" ? (
    <TouchableNativeFeedback onPress={props.onPress}>
      {props.children}
    </TouchableNativeFeedback>
  ) : (
    <TouchableOpacity onPress={props.onPress}>
      {props.children}
    </TouchableOpacity>
  );
};

Touchable.propTypes = {
  onPress: PropTypes.func.isRequired
};

export default Touchable;
