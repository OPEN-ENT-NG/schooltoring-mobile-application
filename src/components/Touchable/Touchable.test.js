import React from "react";

import { shallow } from "enzyme";

import Touchable from "./Touchable";

const onPressCb = jest.fn();
const rendered = shallow(<Touchable onPress={onPressCb} />);

describe("Render", () => {
  test("Touchable should render without crashing", () => {
    expect(rendered).toMatchSnapshot();
  });
});

describe("Interactions", () => {
  test("Press on touchable should trigger onPress prop function", () => {
    rendered.simulate("press");
    expect(onPressCb).toHaveBeenCalled();
  });
});
