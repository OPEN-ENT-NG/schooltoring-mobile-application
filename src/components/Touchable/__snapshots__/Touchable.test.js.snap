// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Touchable should render without crashing 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
/>
`;
